Lib/test/test_zipfile.py

```diff
@@ -2262,6 +2262,29 @@ def test_from_dir(self):
         self.assertEqual(zi.file_size, 0)
 
 
+class ZipInfoWithExtraTest(unittest.TestCase):
+
+    def setUp(self):
+        os.mkdir(TESTFN2)
+
+    def tearDown(self):
+        rmtree(TESTFN2)
+
+    @requires_zlib
+    def test_extra_field(self):
```

Comment:
just curious, is it possible to use Pathlib here?

Reply:
I don't see why not, I've picked `os.path.join` for uniformity just to follow the pattern used in other tests - if Pathlib is preferable option, I can change my code.